typing/typetexp.ml

```diff
@@ -300,7 +291,7 @@ and transl_type_aux env policy styp =
         )
         (List.combine stl args) params;
       let ty_args = List.map (fun ctyp -> ctyp.ctyp_type) args in
-      let ty = Ctype.expand_head env (newconstr path ty_args) in
+      let ty = Ctype.apply ~use_current_level:true env params body ty_args in
```

Comment:
I reviewed the typetexp.ml part of this PR, with Jacques' explanations. It is tricky but I approve of this part of the change.